src/Symfony/Component/Security/Core/Authorization/ExpressionLanguageProvider.php

```diff
@@ -31,9 +32,9 @@ public function getFunctions()
             }),
 
             new ExpressionFunction('is_authenticated', function () {
-                return '$token && !$auth_checker->isGranted("IS_ANONYMOUS")';
+                return '!$token instanceof '.NullToken::class;
             }, function (array $variables) {
-                return $variables['token'] && !$variables['auth_checker']->isGranted('IS_ANONYMOUS');
+                return !$variables['token'] instanceof NullToken;
             }),
```

Comment:
Afaics, this is broken in 5.2+ when using the new system. This should probably be a bugfix.

Reply:
I can't see any evidence that this is indeed a bug